### ISSUE TITLE
Cannot compile templates example

### ISSUE THREAD

**Issue by richardmarston** [SergioBenitez/Rocket#1740]
**Description**

I'm trying to compile the template example as shown here:
https://api.rocket.rs/master/rocket_dyn_templates/struct.Template.html

I have managed to add the missing dependencies and imports but I'm still getting the error:
```bash
error[E0277]: the trait bound `impl rocket::fairing::Fairing: Fairing` is not satisfied
 --> src/main.rs:8:28
  |
8 |     rocket::build().attach(Template::fairing())
  |                            ^^^^^^^^^^^^^^^^^^^ the trait `Fairing` is not implemented for `impl rocket::fairing::Fairing`

error[E0277]: the trait bound `rocket_dyn_templates::Template: Responder<'_, '_>` is not satisfied
  --> src/main.rs:12:15
   |
12 | fn index() -> Template {
   |               ^^^^^^^^ the trait `Responder<'_, '_>` is not implemented for `rocket_dyn_templates::Template`
   |
   = note: required by `route::handler::<impl Outcome<rocket::Response<'o>, Status, rocket::Data<'o>>>::from`

error: aborting due to 2 previous errors
```
**To Reproduce**

main.rs:
```rust
use std::collections::HashMap;
use rocket_dyn_templates::Template;
#[macro_use]
extern crate rocket;
#[launch]
fn rocket() -> _ {
    rocket::build().attach(Template::fairing())
}

#[get("/")]
fn index() -> Template {
    let mut context: HashMap<&str, isize> = HashMap::new();
    Template::render("template-name", &context)
}
```

Cargo.toml:
```toml
[package]
name = "test"
version = "0.1.0"
edition = "2018"

[dependencies]
rocket = { git = "https://github.com/SergioBenitez/Rocket", version = "0.5.0-rc.1", features = [ "json" ] }

[dependencies.rocket_dyn_templates]
version = "0.1.0-rc.1"
features = ["handlebars", "tera"]
```
**Expected Behavior**

I would like to be able to compile the example.

**Environment:**

 - OS Distribution and Kernel: Ubuntu 20.04 
 - Rocket Version: 0.5.0-rc.1

**Additional Context**
I'm new to rust and it's highly likely to be my mistake.

**Comment by SergioBenitez** [comment]
You can never depend on git and a crates.io version of libraries that share types and have it work. They are different versions. Use the published 0.5.0-rc.1. There's no reason to use the version from git, and doing so is wrong as it is not a stable branch.

**Comment by richardmarston** [comment]
Thank you for a very quick response.

The git line was included because I was working with a -dev version before the -rc1 release.
